packages/flutter_tools/lib/src/cache.dart

```diff
@@ -302,8 +302,15 @@ class Cache {
       return;
     }
     assert(_lock == null);
-    final File lockFile =
-      _fileSystem.file(_fileSystem.path.join(flutterRoot!, 'bin', 'cache', 'lockfile'));
+    final Directory cacheDir =
```

Comment:
The cache dir is usually created by the entrypoint script in bin/flutter or bin/dart. What is the usecase for running the tool with a partially set up SDK directly?

Reply:
F

Reply:
> The cache dir is usually created by the entrypoint script in bin/flutter or bin/dart. What is the usecase for running the tool with a partially set up SDK directly?

The use case is running the flutter tools from a cloned source directly, e.g. I was investigating and trying to debug the `flutter` CLI and came across this.

Reply:
Were you running it with a different `dart` than what comes with it? That is the only way you could get here without the cache dir existing

Reply:
I see your point.

I cloned `flutter` project again, and used the `dart-sdk` from the originally installed `master`.

Please close this issue if the user must always run `flutter.bat` whenever a clone is done.

Reply:
This is working as intended, the bootstrap scripts are not optional.